pandas/tests/series/methods/test_replace.py

```diff
@@ -512,3 +512,12 @@ def test_pandas_replace_na(self):
         result = ser.replace(regex_mapping, regex=True)
         exp = pd.Series(["CC", "CC", "CC-REPL", "DD", "CC", "", pd.NA], dtype="string")
         tm.assert_series_equal(result, exp)
+
+    def test_replace_regex_dtype(self):
+        # GH-48644
+        s = pd.Series(["0"])
+        expected = pd.Series([1])
+        result_series_false = s.replace(to_replace="0", value=1, regex=False)
```

Comment:
Simply call result. You can parametrize the test over regex=True/False

Reply:
Done